src/item.cpp

```diff
@@ -8724,11 +8725,12 @@ bool item::use_charges( const itype_id &what, int &qty, std::list<item> &used,
         } else if( e->count_by_charges() ) {
             if( e->typeId() == what ) {
 
+                if( parent != nullptr ) {
+                    parent->contained_where( *e )->unseal();
+                }
+
                 // if can supply excess charges split required off leaving remainder in-situ
                 item obj = e->split( qty );
-                if( parent ) {
-                    parent->on_contents_changed();
-                }
```

Comment:
This does different things from `item_pocket::on_contents_changed` (such as resetting encumberance cache), and shouldn't be removed.

Also, please move the `unseal` call down here and change it to `parent->contained_where( *e )->on_contents_changed();` for consistency with the other unsealing code.

Reply:
Done.